src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/CockroachDBTranslator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.feedzai.commons.sql.abstraction.engine.impl;
+
+import com.feedzai.commons.sql.abstraction.ddl.DbColumn;
+import com.feedzai.commons.sql.abstraction.dml.Cast;
+import com.feedzai.commons.sql.abstraction.dml.Expression;
+import com.feedzai.commons.sql.abstraction.dml.RepeatDelimiter;
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineRuntimeException;
+import com.feedzai.commons.sql.abstraction.engine.OperationNotSupportedRuntimeException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.VARCHAR_SIZE;
+import static java.lang.String.format;
+
+/**
+ * Provides SQL translation for CockroachDB.
+ *
+ * @author Mário Pereira (mario.arzileiro@feedzai.com)
+ * @since 2.5.0
+ */
+public class CockroachDBTranslator extends PostgreSqlTranslator {
+
+    @Override
+    public String translate(final DbColumn c) {
+        switch (c.getDbColumnType()) {
+            case BOOLEAN:
+                return "BOOLEAN";
+
+            case DOUBLE:
+                return "DOUBLE PRECISION";
+
+            case INT:
+                if (c.isAutoInc()) {
```

Comment:
Condition seems unnecessary since INT4 is returned any way.

Reply:
Good point :smile: 
Maybe later we will want to make a distinction here if we just want a generated unique number, but if we want an autoinc in CockroachDB similar to the other engines, then the translation here is the same, autoinc or not.